### PR TITLE
[diffusion] Support MiniMax-H3 PDD Acc LoRA (8-NFE)

### PR DESCRIPTION
**Pull request by niehen6174**
## Motivation

[MiniMax's 2026-09-14 ecosystem note](https://x.com/MiniMax_AI/status/2099381310733328784) lists the acceleration stack the community has been building on H3: FastH3, Sol-H3, VDN, LightX2V Turbo, and PDD Acc-LoRAs. SGLang already serves the first four in-tree:

| Scheme | What it is | SGLang |
|---|---|---|
| FastH3 | 4-step DMD2 + VSA-H3 | [#37480](https://github.com/sgl-project/sglang/pull/37480) (merged weights + `video_sparse_attn_h3`); LoRA path [#37139](https://github.com/sgl-project/sglang/pull/37139) |
| Sol-H3 | training-free sparse attention | [#33702](https://github.com/sgl-project/sglang/pull/33702) (`sol_attn`) |
| VDN-H3 | hybrid window + Video Delta linear attn, 8-NFE | [#37903](https://github.com/sgl-project/sglang/pull/37903) |
| LightX2V Turbo | 4/8-step PEFT Turbo LoRAs | [#34359](https://github.com/sgl-project/sglang/pull/34359) (supersedes [#34258](https://github.com/sgl-project/sglang/pull/34258)); header alpha [#35952](https://github.com/sgl-project/sglang/pull/35952) |

PDD is the remaining official recipe. [Parallel Decoding Distillation](https://arxiv.org/abs/2607.26004) (Shaul, Liu, Vahdat, Berner; arXiv:2607.26004) is a recent NVIDIA trajectory-based distill: each network evaluation predicts a *block* of denoising steps (the mean velocity over a trained Δt interval) instead of one Euler step. Unlike VSD / adversarial few-step students it does not need JVPs or a discriminator, keeps a variable NFE, and the paper reports less diversity / motion collapse at 4–8 NFE.

Alibaba PAI released the official H3 port as [alibaba-pai/MiniMax-H3-Acc-LoRAs](https://huggingface.co/alibaba-pai/MiniMax-H3-Acc-LoRAs) (`rank=64`, `alpha=64`, `pdd_num_steps=32`, `pdd_block_size=4` → **8 NFE**). The files are **not** PEFT-only: a backbone LoRA is paired with a length-32 bank of `proj_out` / `audio_proj_out` heads. A plain `--lora-path` drops the bank and silently loses the distill. This PR adds the same contract to SGLang so the official Acc-LoRA can be served like any other H3 adapter.

On the same 768p / 5s / seed-1000 T2VA prompt, 8-NFE PDD stays on the base-H3 fox-in-snow trajectory with a much smaller quality gap than the more aggressive 4-step Turbo / FastH3 students, at ~6× fewer DiT forwards than the stock 50-step schedule.

## Changes

- **`pdd_lora.py`.** Split an official / Comfy `set_weight` Acc file into backbone LoRA + `[N, out, in]` video/audio head banks. Fuse one `block_size`-wide, Δt-weighted mean head per Euler step (VideoX-Fun / ComfyUI math). TP-local `pdd_linear` shards the fused out-features.
- **`LoRAPipeline`.** `extract_pdd_payload` before `normalize_lora_state_dict`; install the bank on `transformer.final_layer` after the backbone merge; clear it on deactivate. Ordinary LoRAs are unchanged (`extract_pdd_payload` returns `None` when there is no bank).
- **`MiniMaxH3FinalLayer` / denoise loop.** `_project_outputs` uses the armed PDD head when a plan is set; `arm_pdd_step(step)` is called each loop iteration. Stock `simple` at `num_inference_steps=9` (8 DiT forwards) with shifts 12 / 3 lands on the 32-grid block boundaries.
- **`_accepts_mxfp8_input`.** Unwrap `base_layer` so a LoRA-wrapped `RowParallelLinear` does not crash the new mxfp8 probe (any H3 LoRA on current main, not PDD-specific).

Released metadata is `pdd_num_steps=32`, `pdd_block_size=4` (8 NFE). `final_layer.video_out` / `audio_out` stay on the bank, so the 258/266 backbone-layer miss on those two heads is expected.

## E2E (1× H200, T2VA 768p 16:9, 5s, seed 1000, video=12 / audio=3)

Same prompt / seed / resolution / scheduler. PDD uses the official FL2VA Acc-8Step file; base is MiniMax-H3 FL2VA with no adapter. Times include encode + denoise + decode + save; wall also includes load.

prompt: `"integrated_multimodal_description: A red fox runs through fresh snow at dawn. overall_soundscape: Fast pawsteps in snow, winter wind, and distant birds."`

| Case | Steps (NFE) | Denoise | s/step | Pixel | Wall | Peak VRAM |
|---|---:|---:|---:|---:|---:|---:|
| PDD Acc LoRA | 8 | 46.3s | 5.8s | 79.5s | 194s | 78.9 GB |
| Official H3 FA | 50 | 263.3s | 5.4s | 294.0s | 368s | 78.7 GB |

Denoise is **5.7×** (step ratio 50/8 ≈ 6.3). Pixel time is **3.7×**. Both clips stay on the same scene (fox, dawn, snow, run-away). PDD is a tighter, cooler crop; the 50-step base is warmer and opens onto trees / houses later. That is the expected distill tradeoff, not a loader bug.

**PDD Acc 8 NFE** 

https://github.com/user-attachments/assets/574b12a9-ed48-47e1-9c4b-8a5ce64baf33

**Official H3 50-step** 

https://github.com/user-attachments/assets/ecf32741-1463-487a-ab58-cd8cf0677790

## Examples

Adapter: [alibaba-pai/MiniMax-H3-Acc-LoRAs](https://huggingface.co/alibaba-pai/MiniMax-H3-Acc-LoRAs) `MiniMax-H3-FL2VA-Acc-8Step.safetensors`. FL2VA partition (`t2va` / `fl2va`). Use `--num-inference-steps 9` (9 sigma points / 8 DiT forwards). Do **not** stack with Turbo / FastH3 / `quality: "high"`. `--lora-alpha 64 --lora-scale 1.0`.

```bash
sglang generate \
  --model-path MiniMaxAI/MiniMax-H3 \
  --model-variant fl2va \
  --lora-path alibaba-pai/MiniMax-H3-Acc-LoRAs \
  --lora-weight-name MiniMax-H3-FL2VA-Acc-8Step.safetensors \
  --lora-nickname pdd_acc \
  --lora-alpha 64 --lora-scale 1.0 \
  --lora-merge-mode merge \
  --attention-backend fa \
  --num-inference-steps 9 --seed 1000 \
  --prompt "integrated_multimodal_description: A red fox runs through fresh snow at dawn. overall_soundscape: Fast pawsteps in snow, winter wind, and distant birds." \
  --save-output
```

Ref2VA uses `MiniMax-H3-Ref2VA-Acc-8Step.safetensors` with `--model-variant ref2va`.

## Test plan

- [x] `pytest` `python/sglang/multimodal_gen/test/unit/test_pdd_lora.py` + `test_minimax_h3_denoise_loop.py` (19)
- [x] Official FL2VA Acc file parses as N=32 / block=4 / nfe=8, bank `(32, 96, 5376)` / `(32, 32, 5376)`
- [x] T2VA 768p 8 NFE PDD (seed 1000)
- [x] T2VA 768p 50-step base, same prompt / seed
- [ ] Cookbook recipe (can follow in this PR or a docs follow-up)

---
### CI States

Latest PR Test (Base): :x: [Run #35677698569](https://github.com/sgl-project/sglang/actions/runs/35677698569)
Latest PR Test (Extra): :x: [Run #35677698387](https://github.com/sgl-project/sglang/actions/runs/35677698387)
Latest PR Test (AMD ROCm 10): :x: [Run #35677698570](https://github.com/sgl-project/sglang/actions/runs/35677698570)

### CODE REVIEW

**Review by mickqian — COMMENTED** [review]
CI at this head is blocked by check-changes requiring main commit ddd4600197a4: https://github.com/sgl-project/sglang/actions/runs/35677698569/job/106587562455 . Rerunning unchanged code will not clear this gate. The 14:30 UTC snapshot reported conflicts; the latest mergeability query is UNKNOWN, not confirmation they are resolved. Please merge current main and resolve conflicts before final CI. I do not have fork push access.
